{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from cyvcf2 import VCF\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"set path of VCF file and index the file using tabix\"\"\"\n",
    "\n",
    "vcf_path = '//Volumes/Samsung_T5/ncVEP/VCFs/chr21/gnomad.genomes.v4.1.sites.chr21.vcf.bgz'\n",
    "\n",
    "# use tabix to index the gzipped VCF file\n",
    "os.system('tabix -p vcf ' + vcf_path)\n",
    "\n",
    "vcf = VCF(vcf_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr21\t5030082\t.\tG\tA\t.\tAC0;AS_VQSR\tAC=0;AN=2;AF=0;AC_XX=0;AF_XX=0;AN_XX=2;nhomalt_XX=0;AC_XY=0;AN_XY=0;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AN_afr_XX=0;nhomalt_afr_XX=0;AC_afr_XY=0;AN_afr_XY=0;nhomalt_afr_XY=0;AC_afr=0;AN_afr=0;nhomalt_afr=0;AC_ami_XX=0;AN_ami_XX=0;nhomalt_ami_XX=0;AC_ami_XY=0;AN_ami_XY=0;nhomalt_ami_XY=0;AC_ami=0;AN_ami=0;nhomalt_ami=0;AC_amr_XX=0;AN_amr_XX=0;nhomalt_amr_XX=0;AC_amr_XY=0;AN_amr_XY=0;nhomalt_amr_XY=0;AC_amr=0;AN_amr=0;nhomalt_amr=0;AC_asj_XX=0;AN_asj_XX=0;nhomalt_asj_XX=0;AC_asj_XY=0;AN_asj_XY=0;nhomalt_asj_XY=0;AC_asj=0;AN_asj=0;nhomalt_asj=0;AC_eas_XX=0;AF_eas_XX=0;AN_eas_XX=2;nhomalt_eas_XX=0;AC_eas_XY=0;AN_eas_XY=0;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0;AN_eas=2;nhomalt_eas=0;AC_fin_XX=0;AN_fin_XX=0;nhomalt_fin_XX=0;AC_fin_XY=0;AN_fin_XY=0;nhomalt_fin_XY=0;AC_fin=0;AN_fin=0;nhomalt_fin=0;AC_mid_XX=0;AN_mid_XX=0;nhomalt_mid_XX=0;AC_mid_XY=0;AN_mid_XY=0;nhomalt_mid_XY=0;AC_mid=0;AN_mid=0;nhomalt_mid=0;AC_nfe_XX=0;AN_nfe_XX=0;nhomalt_nfe_XX=0;AC_nfe_XY=0;AN_nfe_XY=0;nhomalt_nfe_XY=0;AC_nfe=0;AN_nfe=0;nhomalt_nfe=0;AC_raw=4;AF_raw=0.000136305;AN_raw=29346;nhomalt_raw=2;AC_remaining_XX=0;AN_remaining_XX=0;nhomalt_remaining_XX=0;AC_remaining_XY=0;AN_remaining_XY=0;nhomalt_remaining_XY=0;AC_remaining=0;AN_remaining=0;nhomalt_remaining=0;AC_sas_XX=0;AN_sas_XX=0;nhomalt_sas_XX=0;AC_sas_XY=0;AN_sas_XY=0;nhomalt_sas_XY=0;AC_sas=0;AN_sas=0;nhomalt_sas=0;faf95=0;faf95_afr=0;faf95_amr=0;faf95_eas=0;faf95_nfe=0;faf95_sas=0;faf99=0;faf99_afr=0;faf99_amr=0;faf99_eas=0;faf99_nfe=0;faf99_sas=0;age_hist_het_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=.;MQ=30.6084;QUALapprox=261;QD=32.625;SOR=4.407;VarDP=8;AS_FS=.;AS_MQ=30.6084;AS_QUALapprox=261;AS_QD=32.625;AS_SB_TABLE=0,0|8,0;AS_SOR=4.40672;AS_VarDP=8;inbreeding_coeff=1;AS_culprit=AS_SOR;AS_VQSLOD=-15.2842;allele_type=snv;n_alt_alleles=1;variant_type=snv;segdup;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;gq_hist_all_bin_freq=0|0|0|0|1|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|1|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_all_n_larger=0;ab_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;cadd_raw_score=0.280444;cadd_phred=4.061;phylop=0.548;VRS_Allele_IDs=ga4gh:VA.9ENogSruYatevtFUuE3mJQEpNswN2RV5,ga4gh:VA.EJjRG7fXPmNk28uI-gRIb8kblxXt8XV7;VRS_Starts=5030081,5030081;VRS_Ends=5030082,5030082;VRS_States=G,A;vep=A|intron_variant|MODIFIER||ENSG00000277117|Transcript|ENST00000612610|protein_coding||4/6|ENST00000612610.4:c.697+1857G>A|||||||1||1||SNV||||||1|A2||ENSP00000483732||Ensembl|||||||||||||,A|intron_variant|MODIFIER||ENSG00000277117|Transcript|ENST00000620481|protein_coding||3/5|ENST00000620481.4:c.346+1857G>A|||||||1||1||SNV||||||5|||ENSP00000484302||Ensembl|||||||||||||,A|intron_variant|MODIFIER||ENSG00000277117|Transcript|ENST00000623795|protein_coding||3/5|ENST00000623795.1:c.346+1857G>A|||||||1||1||SNV||||||2|||ENSP00000485649||Ensembl|||||||||||||,A|intron_variant&NMD_transcript_variant|MODIFIER||ENSG00000277117|Transcript|ENST00000623903|nonsense_mediated_decay||4/6|ENST00000623903.3:c.*311+1857G>A|||||||1||1||SNV||||||2|||ENSP00000485557||Ensembl|||||||||||||,A|intron_variant|MODIFIER||ENSG00000277117|Transcript|ENST00000623960|protein_coding||4/6|ENST00000623960.4:c.697+1857G>A|||||||1||1||SNV|||YES|NM_001363770.2||1|P2|CCDS86973.1|ENSP00000485129||Ensembl|||||||||||||,A|intron_variant|MODIFIER|LOC102723996|102723996|Transcript|NM_001363770.2|protein_coding||4/6|NM_001363770.2:c.697+1857G>A|||||||1||1||SNV|EntrezGene||YES|ENST00000623960.4|||||NP_001350699.1||RefSeq|||||||||||||,A|intron_variant|MODIFIER|LOC102723996|102723996|Transcript|XM_006723899.2|protein_coding||4/5|XM_006723899.2:c.697+1857G>A|||||||1||1||SNV|EntrezGene||||||||XP_006723962.1||RefSeq|||||||||||||,A|intron_variant|MODIFIER|LOC102723996|102723996|Transcript|XM_011546078.2|protein_coding||4/6|XM_011546078.2:c.697+1857G>A|||||||1||1||SNV|EntrezGene||||||||XP_011544380.1||RefSeq|||||||||||||,A|intron_variant|MODIFIER|LOC102723996|102723996|Transcript|XM_011546079.1|protein_coding||4/6|XM_011546079.1:c.697+1857G>A|||||||1||1||SNV|EntrezGene||||||||XP_011544381.1||RefSeq|||||||||||||\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#view variant at a specific position\n",
    "position = 5030082\n",
    "\n",
    "for variant in vcf('chr21:'+str(position)+'-'+str(position)):\n",
    "    print(variant)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\n",
      "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
      "##hailversion=0.2.122-be9d88a80695\n",
      "##FILTER=<ID=AC0,Description=\"Allele count is zero after filtering out low-confidence genotypes (GQ < 20; DP < 10; and AB < 0.2 for het calls)\">\n",
      "##FILTER=<ID=AS_VQSR,Description=\"Failed VQSR filtering thresholds of -2.502 for SNPs and -0.7156 for indels\">\n",
      "##FILTER=<ID=InbreedingCoeff,Description=\"Inbreeding coefficient < -0.3\">\n",
      "##INFO=<ID=AC,Number=A,Type=Integer,Description=\"Alternate allele count\">\n",
      "##INFO=<ID=AN,Number=1,Type=Integer,Description=\"Total number of alleles\">\n",
      "##INFO=<ID=AF,Number=A,Type=Float,Description=\"Alternate allele frequency\">\n",
      "##INFO=<ID=grpmax,Number=A,Type=String,Description=\"Genetic ancestry group with the maximum allele frequency\">\n",
      "##INFO=<ID=fafmax_faf95_max,Number=A,Type=Float,Description=\"Maximum filtering allele frequency (using Poisson 95% CI) across genetic ancestry groups\">\n",
      "##INFO=<ID=fafmax_faf95_max_gen_anc,Number=A,Type=String,Description=\"Genetic ancestry group with maximum filtering allele frequency (using Poisson 95% CI)\">\n",
      "##INFO=<ID=AC_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples\">\n",
      "##INFO=<ID=AF_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples\">\n",
      "##INFO=<ID=AN_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples\">\n",
      "##INFO=<ID=nhomalt_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples\">\n",
      "##INFO=<ID=AC_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples\">\n",
      "##INFO=<ID=AF_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples\">\n",
      "##INFO=<ID=AN_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples\">\n",
      "##INFO=<ID=nhomalt_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples\">\n",
      "##INFO=<ID=nhomalt,Number=A,Type=Integer,Description=\"Count of homozygous individuals\">\n",
      "##INFO=<ID=AC_afr_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AF_afr_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AN_afr_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_afr_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AC_afr_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AF_afr_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AN_afr_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_afr_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AC_afr,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AF_afr,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AN_afr,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_afr,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=AC_ami_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AF_ami_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AN_ami_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_ami_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AC_ami_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AF_ami_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AN_ami_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_ami_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AC_ami,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AF_ami,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AN_ami,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_ami,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the Amish genetic ancestry group\">\n",
      "##INFO=<ID=AC_amr_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AF_amr_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AN_amr_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_amr_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AC_amr_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AF_amr_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AN_amr_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_amr_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AC_amr,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AF_amr,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AN_amr,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_amr,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=AC_asj_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AF_asj_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AN_asj_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_asj_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AC_asj_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AF_asj_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AN_asj_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_asj_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AC_asj,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AF_asj,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AN_asj,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_asj,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the Ashkenazi Jewish genetic ancestry group\">\n",
      "##INFO=<ID=AC_eas_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AF_eas_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AN_eas_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_eas_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AC_eas_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AF_eas_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AN_eas_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_eas_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AC_eas,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AF_eas,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AN_eas,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_eas,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=AC_fin_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AF_fin_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AN_fin_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_fin_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AC_fin_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AF_fin_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AN_fin_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_fin_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AC_fin,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AF_fin,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AN_fin,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_fin,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the Finnish genetic ancestry group\">\n",
      "##INFO=<ID=AC_mid_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AF_mid_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AN_mid_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_mid_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AC_mid_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AF_mid_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AN_mid_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_mid_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AC_mid,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AF_mid,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AN_mid,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_mid,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the Middle Eastern genetic ancestry group\">\n",
      "##INFO=<ID=AC_nfe_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AF_nfe_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AN_nfe_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_nfe_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AC_nfe_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AF_nfe_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AN_nfe_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_nfe_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AC_nfe,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AF_nfe,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AN_nfe,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_nfe,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=AC_raw,Number=A,Type=Integer,Description=\"Alternate allele count for samples, before removing low-confidence genotypes\">\n",
      "##INFO=<ID=AF_raw,Number=A,Type=Float,Description=\"Alternate allele frequency in samples, before removing low-confidence genotypes\">\n",
      "##INFO=<ID=AN_raw,Number=1,Type=Integer,Description=\"Total number of alleles in samples, before removing low-confidence genotypes\">\n",
      "##INFO=<ID=nhomalt_raw,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples, before removing low-confidence genotypes\">\n",
      "##INFO=<ID=AC_remaining_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AF_remaining_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AN_remaining_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_remaining_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AC_remaining_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AF_remaining_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AN_remaining_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_remaining_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AC_remaining,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AF_remaining,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AN_remaining,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_remaining,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the Remaining individuals genetic ancestry group\">\n",
      "##INFO=<ID=AC_sas_XX,Number=A,Type=Integer,Description=\"Alternate allele count for XX samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AF_sas_XX,Number=A,Type=Float,Description=\"Alternate allele frequency in XX samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AN_sas_XX,Number=1,Type=Integer,Description=\"Total number of alleles in XX samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_sas_XX,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XX samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AC_sas_XY,Number=A,Type=Integer,Description=\"Alternate allele count for XY samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AF_sas_XY,Number=A,Type=Float,Description=\"Alternate allele frequency in XY samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AN_sas_XY,Number=1,Type=Integer,Description=\"Total number of alleles in XY samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_sas_XY,Number=A,Type=Integer,Description=\"Count of homozygous individuals in XY samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AC_sas,Number=A,Type=Integer,Description=\"Alternate allele count for samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AF_sas,Number=A,Type=Float,Description=\"Alternate allele frequency in samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AN_sas,Number=1,Type=Integer,Description=\"Total number of alleles in samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=nhomalt_sas,Number=A,Type=Integer,Description=\"Count of homozygous individuals in samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=AC_grpmax,Number=A,Type=Integer,Description=\"Allele count in the genetic ancestry group with the maximum allele frequency\">\n",
      "##INFO=<ID=AF_grpmax,Number=A,Type=Float,Description=\"Maximum allele frequency across genetic ancestry groups\">\n",
      "##INFO=<ID=AN_grpmax,Number=A,Type=Integer,Description=\"Total number of alleles in the genetic ancestry group with the maximum allele frequency\">\n",
      "##INFO=<ID=nhomalt_grpmax,Number=A,Type=Integer,Description=\"Count of homozygous individuals in the genetic ancestry group with the maximum allele frequency\">\n",
      "##INFO=<ID=faf95_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XX samples in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XY samples in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI)\">\n",
      "##INFO=<ID=faf95_afr_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XX samples in the African/African-American genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_afr_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XY samples in the African/African-American genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_afr,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=faf95_amr_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XX samples in the Admixed American genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_amr_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XY samples in the Admixed American genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_amr,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=faf95_eas_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XX samples in the East Asian genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_eas_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XY samples in the East Asian genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_eas,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=faf95_nfe_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XX samples in the Non-Finnish European genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_nfe_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XY samples in the Non-Finnish European genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_nfe,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=faf95_sas_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XX samples in the South Asian genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_sas_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for XY samples in the South Asian genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf95_sas,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 95% CI) for samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=faf99_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XX samples in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XY samples in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI)\">\n",
      "##INFO=<ID=faf99_afr_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XX samples in the African/African-American genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_afr_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XY samples in the African/African-American genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_afr,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for samples in the African/African-American genetic ancestry group\">\n",
      "##INFO=<ID=faf99_amr_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XX samples in the Admixed American genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_amr_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XY samples in the Admixed American genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_amr,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for samples in the Admixed American genetic ancestry group\">\n",
      "##INFO=<ID=faf99_eas_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XX samples in the East Asian genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_eas_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XY samples in the East Asian genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_eas,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for samples in the East Asian genetic ancestry group\">\n",
      "##INFO=<ID=faf99_nfe_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XX samples in the Non-Finnish European genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_nfe_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XY samples in the Non-Finnish European genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_nfe,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for samples in the Non-Finnish European genetic ancestry group\">\n",
      "##INFO=<ID=faf99_sas_XX,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XX samples in the South Asian genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_sas_XY,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for XY samples in the South Asian genetic ancestry group in non-PAR regions of sex chromosomes only\">\n",
      "##INFO=<ID=faf99_sas,Number=A,Type=Float,Description=\"Filtering allele frequency (using Poisson 99% CI) for samples in the South Asian genetic ancestry group\">\n",
      "##INFO=<ID=fafmax_faf99_max,Number=A,Type=Float,Description=\"Maximum filtering allele frequency (using Poisson 99% CI) across genetic ancestry groups\">\n",
      "##INFO=<ID=fafmax_faf99_max_gen_anc,Number=A,Type=String,Description=\"Genetic ancestry group with maximum filtering allele frequency (using Poisson 99% CI)\">\n",
      "##INFO=<ID=age_hist_het_bin_freq,Number=A,Type=String,Description=\"Histogram of ages of heterozygous individuals; bin edges are: 30.0|35.0|40.0|45.0|50.0|55.0|60.0|65.0|70.0|75.0|80.0; total number of individuals of any genotype bin: 1332|1401|1642|2949|4283|3601|3282|2909|1955|1202\">\n",
      "##INFO=<ID=age_hist_het_n_smaller,Number=A,Type=Integer,Description=\"Count of age values falling below lowest histogram bin edge for heterozygous individuals\">\n",
      "##INFO=<ID=age_hist_het_n_larger,Number=A,Type=Integer,Description=\"Count of age values falling above highest histogram bin edge for heterozygous individuals\">\n",
      "##INFO=<ID=age_hist_hom_bin_freq,Number=A,Type=String,Description=\"Histogram of ages of homozygous alternate individuals; bin edges are: 30.0|35.0|40.0|45.0|50.0|55.0|60.0|65.0|70.0|75.0|80.0; total number of individuals of any genotype bin: 1332|1401|1642|2949|4283|3601|3282|2909|1955|1202\">\n",
      "##INFO=<ID=age_hist_hom_n_smaller,Number=A,Type=Integer,Description=\"Count of age values falling below lowest histogram bin edge for homozygous alternate individuals\">\n",
      "##INFO=<ID=age_hist_hom_n_larger,Number=A,Type=Integer,Description=\"Count of age values falling above highest histogram bin edge for homozygous alternate individuals\">\n",
      "##INFO=<ID=FS,Number=1,Type=Float,Description=\"Phred-scaled p-value of Fisher's exact test for strand bias\">\n",
      "##INFO=<ID=MQ,Number=1,Type=Float,Description=\"Root mean square of the mapping quality of reads across all samples\">\n",
      "##INFO=<ID=MQRankSum,Number=1,Type=Float,Description=\"Z-score from Wilcoxon rank sum test of alternate vs. reference read mapping qualities\">\n",
      "##INFO=<ID=QUALapprox,Number=1,Type=Integer,Description=\"Sum of PL[0] values; used to approximate the QUAL score\">\n",
      "##INFO=<ID=QD,Number=1,Type=Float,Description=\"Variant call confidence normalized by depth of sample reads supporting a variant\">\n",
      "##INFO=<ID=ReadPosRankSum,Number=1,Type=Float,Description=\"Z-score from Wilcoxon rank sum test of alternate vs. reference read position bias\">\n",
      "##INFO=<ID=SOR,Number=1,Type=Float,Description=\"Strand bias estimated by the symmetric odds ratio test\">\n",
      "##INFO=<ID=VarDP,Number=1,Type=Integer,Description=\"Depth over variant genotypes (does not include depth of reference samples)\">\n",
      "##INFO=<ID=monoallelic,Number=0,Type=Flag,Description=\"All samples are homozygous alternate for the variant\">\n",
      "##INFO=<ID=only_het,Number=0,Type=Flag,Description=\"All samples are heterozygous for the variant\">\n",
      "##INFO=<ID=transmitted_singleton,Number=0,Type=Flag,Description=\"Variant was a callset-wide doubleton that was transmitted within a family from a parent to a child (i.e., a singleton amongst unrelated samples in cohort)\">\n",
      "##INFO=<ID=AS_FS,Number=A,Type=Float,Description=\"Allele-specific phred-scaled p-value of Fisher's exact test for strand bias\">\n",
      "##INFO=<ID=AS_MQ,Number=A,Type=Float,Description=\"Allele-specific root mean square of the mapping quality of reads across all samples\">\n",
      "##INFO=<ID=AS_MQRankSum,Number=A,Type=Float,Description=\"Allele-specific z-score from Wilcoxon rank sum test of alternate vs. reference read mapping qualities\">\n",
      "##INFO=<ID=AS_pab_max,Number=A,Type=Float,Description=\"Maximum p-value over callset for binomial test of observed allele balance for a heterozygous genotype, given expectation of 0.5\">\n",
      "##INFO=<ID=AS_QUALapprox,Number=A,Type=Integer,Description=\"Allele-specific sum of PL[0] values; used to approximate the QUAL score\">\n",
      "##INFO=<ID=AS_QD,Number=A,Type=Float,Description=\"Allele-specific variant call confidence normalized by depth of sample reads supporting a variant\">\n",
      "##INFO=<ID=AS_ReadPosRankSum,Number=A,Type=Float,Description=\"Allele-specific z-score from Wilcoxon rank sum test of alternate vs. reference read position bias\">\n",
      "##INFO=<ID=AS_SB_TABLE,Number=.,Type=String,Description=\"Allele-specific forward/reverse read counts for strand bias tests\">\n",
      "##INFO=<ID=AS_SOR,Number=A,Type=Float,Description=\"Allele-specific strand bias estimated by the symmetric odds ratio test\">\n",
      "##INFO=<ID=AS_VarDP,Number=A,Type=Integer,Description=\"Allele-specific depth over variant genotypes (does not include depth of reference samples)\">\n",
      "##INFO=<ID=inbreeding_coeff,Number=A,Type=Float,Description=\"Inbreeding coefficient, the excess heterozygosity at a variant site, computed as 1 - (the number of heterozygous genotypes)/(the number of heterozygous genotypes expected under Hardy-Weinberg equilibrium)\">\n",
      "##INFO=<ID=AS_culprit,Number=A,Type=String,Description=\"Allele-specific worst-performing annotation in the VQSR Gaussian mixture model\">\n",
      "##INFO=<ID=AS_VQSLOD,Number=A,Type=Float,Description=\"Allele-specific log-odds ratio of being a true variant versus being a false positive under the trained VQSR Gaussian mixture model\">\n",
      "##INFO=<ID=negative_train_site,Number=0,Type=Flag,Description=\"Variant was used to build the negative training set of low-quality variants for VQSR\">\n",
      "##INFO=<ID=positive_train_site,Number=0,Type=Flag,Description=\"Variant was used to build the positive training set of high-quality variants for VQSR\">\n",
      "##INFO=<ID=allele_type,Number=1,Type=String,Description=\"Allele type (snv, insertion, deletion, or mixed)\">\n",
      "##INFO=<ID=n_alt_alleles,Number=1,Type=Integer,Description=\"Total number of alternate alleles observed at variant locus\">\n",
      "##INFO=<ID=variant_type,Number=1,Type=String,Description=\"Variant type (snv, indel, multi-snv, multi-indel, or mixed)\">\n",
      "##INFO=<ID=was_mixed,Number=0,Type=Flag,Description=\"Variant type was mixed\">\n",
      "##INFO=<ID=lcr,Number=0,Type=Flag,Description=\"Variant falls within a low complexity region\">\n",
      "##INFO=<ID=non_par,Number=0,Type=Flag,Description=\"Variant (on sex chromosome) falls outside a pseudoautosomal region\">\n",
      "##INFO=<ID=segdup,Number=0,Type=Flag,Description=\"Variant falls within a segmental duplication region\">\n",
      "##INFO=<ID=gq_hist_alt_bin_freq,Number=A,Type=String,Description=\"Histogram for GQ in heterozygous individuals calculated on high quality genotypes; bin edges are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100\">\n",
      "##INFO=<ID=gq_hist_all_bin_freq,Number=A,Type=String,Description=\"Histogram for GQ calculated on high quality genotypes; bin edges are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100\">\n",
      "##INFO=<ID=dp_hist_alt_bin_freq,Number=A,Type=String,Description=\"Histogram for DP in heterozygous individuals calculated on high quality genotypes; bin edges are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100\">\n",
      "##INFO=<ID=dp_hist_alt_n_larger,Number=A,Type=Integer,Description=\"Count of DP values falling above highest histogram bin edge DP in heterozygous individuals calculated on high quality genotypes\">\n",
      "##INFO=<ID=dp_hist_all_bin_freq,Number=A,Type=String,Description=\"Histogram for DP calculated on high quality genotypes; bin edges are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100\">\n",
      "##INFO=<ID=dp_hist_all_n_larger,Number=A,Type=Integer,Description=\"Count of DP values falling above highest histogram bin edge DP calculated on high quality genotypes\">\n",
      "##INFO=<ID=ab_hist_alt_bin_freq,Number=A,Type=String,Description=\"Histogram for AB in heterozygous individuals calculated on high quality genotypes; bin edges are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00\">\n",
      "##INFO=<ID=cadd_raw_score,Number=1,Type=Float,Description=\"Raw CADD scores are interpretable as the extent to which the annotation profile for a given variant suggests that the variant is likely to be 'observed' (negative values) vs 'simulated' (positive values). Larger values are more deleterious.\">\n",
      "##INFO=<ID=cadd_phred,Number=1,Type=Float,Description=\"Cadd Phred-like scores ('scaled C-scores') ranging from 1 to 99, based on the rank of each variant relative to all possible 8.6 billion substitutions in the human reference genome. Larger values are more deleterious.\">\n",
      "##INFO=<ID=revel_max,Number=1,Type=Float,Description=\"The maximum REVEL score at a site's MANE Select or canonical transcript. It's an ensemble score for predicting the pathogenicity of missense variants (based on 13 other variant predictors). Scores ranges from 0 to 1. Variants with higher scores are predicted to be more likely to be deleterious.\">\n",
      "##INFO=<ID=spliceai_ds_max,Number=1,Type=Float,Description=\"Illumina's SpliceAI max delta score; interpreted as the probability of the variant being splice-altering.\">\n",
      "##INFO=<ID=pangolin_largest_ds,Number=1,Type=Float,Description=\"Pangolin's largest delta score across 2 splicing consequences, which reflects the probability of the variant being splice-altering\">\n",
      "##INFO=<ID=phylop,Number=1,Type=Float,Description=\"Base-wise conservation score across the 241 placental mammals in the Zoonomia project. Score ranges from -20 to 9.28, and reflects acceleration (faster evolution than expected under neutral drift, assigned negative scores) as well as conservation (slower than expected evolution, assigned positive scores).\">\n",
      "##INFO=<ID=sift_max,Number=1,Type=Float,Description=\"Score reflecting the scaled probability of the amino acid substitution being tolerated, ranging from 0 to 1. Scores below 0.05 are predicted to impact protein function. We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript.\">\n",
      "##INFO=<ID=polyphen_max,Number=1,Type=Float,Description=\"Score that predicts the possible impact of an amino acid substitution on the structure and function of a human protein, ranging from 0.0 (tolerated) to 1.0 (deleterious).  We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript.\">\n",
      "##INFO=<ID=VRS_Allele_IDs,Number=R,Type=String,Description=\"The computed identifiers for the GA4GH VRS Alleles corresponding to the values in the REF and ALT fields\">\n",
      "##INFO=<ID=VRS_Starts,Number=R,Type=Integer,Description=\"Interresidue coordinates used as the location starts for the GA4GH VRS Alleles corresponding to the values in the REF and ALT fields\">\n",
      "##INFO=<ID=VRS_Ends,Number=R,Type=Integer,Description=\"Interresidue coordinates used as the location ends for the GA4GH VRS Alleles corresponding to the values in the REF and ALT fields\">\n",
      "##INFO=<ID=VRS_States,Number=.,Type=String,Description=\"The literal sequence states used for the GA4GH VRS Alleles corresponding to the values in the REF and ALT fields\">\n",
      "##INFO=<ID=vep,Number=.,Type=String,Description=\"Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|ALLELE_NUM|DISTANCE|STRAND|FLAGS|VARIANT_CLASS|SYMBOL_SOURCE|HGNC_ID|CANONICAL|MANE_SELECT|MANE_PLUS_CLINICAL|TSL|APPRIS|CCDS|ENSP|UNIPROT_ISOFORM|SOURCE|DOMAINS|miRNA|HGVS_OFFSET|PUBMED|MOTIF_NAME|MOTIF_POS|HIGH_INF_POS|MOTIF_SCORE_CHANGE|TRANSCRIPTION_FACTORS|LoF|LoF_filter|LoF_flags|LoF_info\">\n",
      "##age_distribution=bin_edges=[30.0, 35.0, 40.0, 45.0, 50.0, 55.0, 60.0, 65.0, 70.0, 75.0, 80.0], bin_freq=[1332, 1401, 1642, 2949, 4283, 3601, 3282, 2909, 1955, 1202], n_smaller=3760, n_larger=438\n",
      "##cadd_version=v1.6\n",
      "##dbsnp_version=b156\n",
      "##gencode_version=Release 39\n",
      "##mane_select_version=v0.95\n",
      "##pangolin_version=v1.3.12,v1.4.4\n",
      "##phylop_version=v2\n",
      "##polyphen_version=2.2.2\n",
      "##revel_version=v1.3\n",
      "##seqrepo_version=2018-11-26\n",
      "##sift_version=5.2.2\n",
      "##spliceai_version=v1.3\n",
      "##vep_version=v105\n",
      "##vrs_schema_version=1.3.0\n",
      "##vrs_python_version=0.8.4\n",
      "##contig=<ID=chr1,length=248956422,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr2,length=242193529,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr3,length=198295559,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr4,length=190214555,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr5,length=181538259,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr6,length=170805979,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr7,length=159345973,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr8,length=145138636,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr9,length=138394717,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr10,length=133797422,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr11,length=135086622,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr12,length=133275309,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr13,length=114364328,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr14,length=107043718,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr15,length=101991189,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr16,length=90338345,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr17,length=83257441,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr18,length=80373285,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr19,length=58617616,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr20,length=64444167,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr21,length=46709983,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chr22,length=50818468,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chrX,length=156040895,assembly=gnomAD_GRCh38>\n",
      "##contig=<ID=chrY,length=57227415,assembly=gnomAD_GRCh38>\n",
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# get the header of the VCF file\n",
    "header = vcf.raw_header\n",
    "print(header)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save only if variant_type is SNV\n",
    "chrom = []\n",
    "pos = []\n",
    "ID = []\n",
    "ref = []\n",
    "alt = []\n",
    "for variant in vcf:\n",
    "    if variant.INFO[\"variant_type\"] == \"snv\":\n",
    "        chrom.append(variant.CHROM)\n",
    "        pos.append(variant.POS)\n",
    "        ID.append(variant.ID)\n",
    "        ref.append(variant.REF)\n",
    "        alt.append(variant.ALT)\n",
    "\n",
    "# create a dataframe from the lists\n",
    "df = pd.DataFrame({'chrom': chrom, 'pos': pos, 'ID': ID, 'ref': ref, 'alt': alt})\n",
    "\n",
    "#convert alt to string\n",
    "df['alt'] = df['alt'].apply(lambda x: str(x[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "#convert pos to string\n",
    "df['pos'] = df['pos'].apply(lambda x: str(x))\n",
    "\n",
    "#if ID is missing, replace with '.'\n",
    "df['ID'] = df['ID'].apply(lambda x: '.' if x is None else x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save bed file\n",
    "df['annot'] = df['chrom'] + '_' + df['pos'] + \";\" +df['ID'] + '_' + df['ref'] + ':' + df['alt']\n",
    "bed_out = df[['chrom', 'pos', 'pos', 'annot']]\n",
    "bed_out.to_csv('/Volumes/Samsung_T5/ncVEP/VCFs/gnomad_chr21_snvs.bed', sep='\\t', header=False, index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "myenv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
